pkg/statsd/filtering.go

```diff
@@ -0,0 +1,43 @@
+package statsd
+
+import (
+	"github.com/atlassian/gostatsd"
+
+	"github.com/spf13/viper"
+)
+
+type Filter struct {
+	MatchMetrics   gostatsd.StringMatchList // Name must match
+	ExcludeMetrics gostatsd.StringMatchList // Name must not match
+	MatchTags      gostatsd.StringMatchList // Any tag must match
+	DropTags       gostatsd.StringMatchList // Any tag matching anything will be dropped
+	DropMetric     bool                     // Drop the entire metric
+	DropHost       bool                     // Clears Hostname if present
+}
+
+// toStringMatch turns a []string in to a []gostatsd.StringMatch
+func toStringMatch(tests []string) []gostatsd.StringMatch {
+	var matches []gostatsd.StringMatch
+	for _, test := range tests {
+		matches = append(matches, gostatsd.NewStringMatch(test))
+	}
+	return matches
+}
+
+// NewFilterFromViper creates a new Filter given a *viper.Viper
+func NewFilterFromViper(name string, v *viper.Viper) Filter {
```

Comment:
`name` is not used?

Reply:
Good spot.  It was at one point, I was going to emit metrics on the name.  I'll ditch it.